docs/launch-profiles.md

```diff
@@ -0,0 +1,65 @@
+
+The launch command specifies which VS component is responsible for interpreting the active launch profile and taking whatever actions are appropriate when "starting" the project. The .NET Project System provides handlers for the "Executable" command (which, as seen in the example, runs a specified .exe) and the "Project" command, which is very similar but automatically chooses the binary produced by the project as the executable. The ASP.NET Core tooling adds handlers for the "IIS" and "IISExpress" commands, which run the project binary within a web server, and there are a number of others.
+
+The important point is that launch profiles can contain arbitrary settings, and how to interpret those settings is entirely up to the command handler. By convention environment variables are stored in a setting named `environmentVariables`, but a particular command handler doesn't have to look for them there, or even support environment variables at all.
+
+New launch commands can be defined by implementing the [`IDebugProfileLaunchTargetsProvider`](https://github.com/dotnet/project-system/blob/main/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/IDebugProfileLaunchTargetsProvider.cs) interface (and optionally `IDebugProfileLaunchTargetsProvider2`, and `IDebugProfileLaunchTargetsProvider3`). The "Project" and "Executable" commands are both handled by the [`ProjectLaunchTargetsProvider`](https://github.com/dotnet/project-system/blob/main/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ProjectLaunchTargetsProvider.cs), for example.
+
+## Features
+
+### MSBuild property/environment variable substitution
+
+The handler for the "Project" and "Executable" commands will automatically replace MSBuild property references (e.g. `$(MyProperty)`) and environment variable references (e.g. `%MY_VAR%`) with the corresponding values. In the above example this allows us to avoid a hard-coded path to devenv.exe, among other things.
+
+This functionality is available to other launch commands through the [`IDebugTokenReplacer`](https://github.com/dotnet/project-system/blob/main/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/IDebugTokenReplacer.cs) interface.
+
+### Command-specific UI
+
+Each launch command can have its own associated UI for updating properties; see the [Project Property Pages documentation](https://github.com/dotnet/project-system/tree/main/docs/repo/property-pages) for more information.
+
+### In-memory profiles and settings
+
+VS supports in-memory profiles that are not committed to the launchSettings.json. For example, if no launchSettings.json is present the .NET Project System will automatically create an in-memory profile with the project's name and the "Project" launch command; this allows running a project without cluttering your project with a trivial launchSettings.json file. Only when the profile is edited through the UI do we create the file and commit the profile.
```

Comment:
Can we cross reference the code that does this here?

Reply:
Oh, yes. Good catch. I went back and added those links for other things, but missed these.